### DISCUSSION TITLE
go-python3 How to instantiate a python class

### DISCUSSION THREAD

**Discussion by djxhhh** [go-python/cpy3#26] [Usage Discussions]
I don't know how to use go-python3 for class initialization

**Comment by christian-korneck** [comment]
one way would be to just write some Python code to instantiate the class and run it with PyRun_SimpleString. You can then use Python C-API functions to find the resulting PyObject in the root dict. Once you have it mapped to a Go variable you can i.e. read attributes or call methods.

Most of this is outlined in this example. Your instantiated Python class is just a PyObject.

https://github.com/christian-korneck/python-go/tree/master/py-bindings/outliers

Another way: To instantiate a Python class the pure C-API way, this might help:

https://stackoverflow.com/a/1147840/4039255
